assets/style.css

```diff
@@ -57,6 +56,15 @@ main {
 	color: inherit;
 }
 
+:is(h1, h2, h3, h4) {
+	--factor: 3;
```

Comment:
This is a magic number for now. I wonder, what if we eat our dog food and use `StyleObserver` to calculate this factor (or `scroll-margin` directly)? 🤔

Reply:
We don't need `StyleObserver` for this. If we wanted to do it "properly", it seems we need a Resize Observer as it depends on the height of the heading, no? But it seems overkill to do that, just specify something in ems that looks right (but I'd assign it to a `--header-height` variable so we know what it is and then add space to that, whereas a random factor obscures that)